File: MLR-spring/src/main/java/com/mlr/entity/RuleExecutionLog.java
package com.mlr.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Table(name = "rule_execution_logs")
@Getter @Setter @NoArgsConstructor
public class RuleExecutionLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime executedAt = LocalDateTime.now();

    private String ruleName;

    private int usersProcessed;

    private int alertsGenerated;

    private Long executionTimeMs; 
}

File: MLR/DAO/Entities/RuleExecutionLog.cs
using System.ComponentModel.DataAnnotations;

namespace MLR.DAO;

public class RuleExecutionLog
{
    [Key]
    public int Id { get; set; }

    public DateTime ExecutedAt { get; set; } = DateTime.UtcNow;

    public int UsersProcessed { get; set; }

    public int AlertsGenerated { get; set; }

    public TimeSpan ExecutionTime { get; set; }
}
